jupyter-js-widgets/src/widget_string.ts

```diff
@@ -385,3 +385,121 @@ class TextView extends LabeledDOMWidgetView {
 
     textbox: HTMLInputElement;
 }
+
+export
+class PasswordModel extends StringModel {
+    defaults() {
+        return _.extend(super.defaults(), {
+            _view_name: 'PasswordView',
+            _model_name: 'PasswordModel'
+        });
+    }
+}
+
+export
+class PasswordView extends LabeledDOMWidgetView {
```

Comment:
(the only issue I could see here is if the element is appended to the DOM before the input is set to password, the default string might flash up on the screen exposed)